### ISSUE TITLE
Handsontable no longer renders columns properly when doing an updateSettings() after the user moves a column

### ISSUE THREAD

**Issue by cgatesman** [handsontable/handsontable#3256]
I was attempting to update my project from 0.18.0 to 0.22.0, and found a wide spread regression.  I tracked it down to being a regression in 0.22.0, specifically.  0.21.0 does not have the issue.

In my project, I have a need to do an `updateSettings()` to redefine the `columns` on the table when the user moves a column.  I found with 0.22.0 (and now 0.23.0, also), this no longer works, and it does not render out the columns correctly.  It appears that perhaps Handsontable is not clearing out the knowledge of moved columns like it propbably should when told to `updateSettings()` with a new `columns` definition.

The resulting effect in my sample, below, is that if you move a column (e.g. column `B`) to the right, nothing changes visually in the table.

Please have a look at my sample, below, and you will find the broken behavior in 0.22.0 and then it will work if you change the version back to 0.21.0 or earlier:

http://jsfiddle.net/cgatesman/u4pt5dvw/

**Comment by cgatesman** [comment]
Please note, in this particular case, I happen to be doing an `updateSettings()` with a column definition that matches the order of the manual move, but you will find that if the `updateSettings()` does any kind of reorder after the manual move, the rendered columns do not match what was defined in the `updateSettings()`.

**Comment by jansiegel** [comment]
@cgatesman, correct me if I'm wrong, but i think it works properly since `0.22.0`:
* `columns` are set as `'a', 'b', 'c', 'd'`, column position array is not changed: `[0, 1, 2, 3]`
* **Action:** User drags the 2nd column after the 3rd column, which modifies the position array to `[0, 2, 1, 3]`
* Your script modifies the `columns` array to `'a', 'c', 'b', 'd'`.

So, after all this, Handsontable takes the modified `columns` array, modifies the column order using the `manualColumnPositions` array, and displays them. The result is not changed, because the column position array "reverts" the changes made to the `columns` array.

To sum it up: Do you need to rearrange the `columns` settings? If you do, the workaround for this is setting `manualColumnMove: []` in your `updateSettings` call.

Please let me know if it helped!

**Comment by cgatesman** [comment]
I would expect that if you do an updateSettings redefining the columns, it should throw away any passed knowledge of manual moves as you are strictly restating where you want each column to be--think of the column redefinition as a manual move via code.

**Comment by cgatesman** [comment]
> To sum it up: Do you need to rearrange the `columns` settings? If you do, the workaround for this is setting `manualColumnMove: []` in your `updateSettings` call.

I didn't originally understand this as it's kind of a new feature in Handsontable.  As it turns out, this is exactly what I needed for my situation.  Thank you very much.
